core/src/main/java/com/opensymphony/xwork2/inject/ContainerImpl.java

```diff
@@ -164,7 +164,8 @@ private boolean isStatic(Member member) {
         public FieldInjector(ContainerImpl container, Field field, String name)
                 throws MissingDependencyException {
             this.field = field;
-            if (!field.isAccessible()) {
+            if ((!Modifier.isPublic(field.getModifiers()) || !Modifier.isPublic(field.getDeclaringClass().getModifiers()))
+                    && !field.isAccessible()) {
```

Comment:
Maybe it would be a good idea to extract this into a method eg.: `isAccessible(field)` instead of repeating the same code three times?

Reply:
💯 refactored as much as possible 👍